app/models/category_list.rb

```diff
@@ -135,7 +135,16 @@ def find_categories
         @options[:parent_category_id].to_i,
       ) if @options[:parent_category_id].present?
 
-    query = self.class.order_categories(query)
+    if SiteSetting.lazy_load_categories
+      query =
+        query.where(
+          "categories.depth = 0 OR (categories.depth != 0 AND categories.position_to_parent <= 5)",
```

Comment:
We need an index on `position_to_parent` for this. We also need an index on `position` to order by it.

Reply:
Yes, I agree. I was surprised to see we do not have an index for `position` even now because we have the `fixed_category_positions` site setting. I assume because the number of categories is low in general.